src/libraries/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs

```diff
@@ -69,6 +69,28 @@ internal BoundedChannelReader(BoundedChannel<T> parent)
 
             public override Task Completion => _parent._completion.Task;
 
+            public override bool CanCount => true;
+
+            public override int Count
+            {
+                get
+                {
+                    BoundedChannel<T> parent = _parent;
+                    lock (parent.SyncObj)
```

Comment:
do we need to take a lock? I know this will report more accurate count but we are not promising this prompt will always be accurate anyway.

Reply:
Technically probably not (we would need to remove the AssertInvariants), as long as the implementation of the internal `Dequeue<T>.Count` doesn't change.

Reply:
That said, I think we should keep the lock, and only consider removing it if it proves problematic.